App: encode primary key

joselcvarela opened the pull request:
Since primary keys could be manually entered strings,
we should encode those in order to prevent accessing
inexistent routes.
E.g.
Document with primary key: 'clients/John Doe'

Fixes https://github.com/directus/directus/issues/4896

joselcvarela committed:
app: encode primary key
Since primary keys could be manually entered strings,
we should encode those in order to prevent accessing
inexistent routes.
E.g.
Document with primary key: 'clients/John Doe'

rijkvanzanten commented:
@joselcvarela There's a couple more places where we'd have to do this, like:

https://github.com/directus/directus/blob/d6f2a96bd670365fc426827c6783c6bd6e2002a6/app/src/layouts/tabular/tabular.vue#L540

joselcvarela committed:
fixup! app: encode primary key Since primary keys could be manually entered strings, we should encode those in order to prevent accessing inexistent routes. E.g. Document with primary key: 'clients/John Doe'

joselcvarela commented:
@rijkvanzanten in first attempt I just replaced where I experienced problems.
But with the latest commit I have replaced where `/collections/` and `/items/` were found.
All of those change might not be necessary but I think it would not hurt if they exist.

rijkvanzanten committed:
Merge branch 'main' into encode-primary-key

rijkvanzanten committed:
Merge branch 'main' into encode-primary-key
